service/api/service_impl.go

```diff
@@ -149,6 +151,111 @@ func (s serviceImpl) PublishToLocalQueue(
+		Namespace: request.GetNamespace(),
+		ProcessId: request.GetProcessId(),
+	})
+	if err != nil {
+		return nil, s.handleUnknownError(err)
+	}
+
+	if latestPrcExe.NotExists || latestPrcExe.Status != data_models.ProcessExecutionStatusRunning {
+		return nil, NewErrorWithStatus(http.StatusNotFound, "Process does not exist")
+	}
+
+	iwfWorkerBaseUrl := urlautofix.FixWorkerUrl(latestPrcExe.WorkerUrl)
+	apiClient := xdbapi.NewAPIClient(&xdbapi.Configuration{
+		Servers: []xdbapi.ServerConfiguration{
+			{
+				URL: iwfWorkerBaseUrl,
+			},
+		},
+	})
+
+	loadGlobalAttrResp, err := s.store.LoadGlobalAttributes(ctx, data_models.LoadGlobalAttributesRequest{
+		TableConfig: *latestPrcExe.GlobalAttributeConfig,
+		Request:     request.GetLoadGlobalAttributesRequest(),
+	})
+	if err != nil {
+		return nil, s.handleUnknownError(err)
+	}
+
+	workerApiCtx, cancF := s.createContextWithTimeoutForRpc(ctx, request.GetTimeoutSeconds())
+	defer cancF()
+
+	req := apiClient.DefaultAPI.ApiV1XdbWorkerProcessRpcPost(workerApiCtx)
+	resp, httpResp, err := req.ProcessRpcWorkerRequest(
+		xdbapi.ProcessRpcWorkerRequest{
+			Context: xdbapi.Context{
+				ProcessId:               request.GetProcessId(),
+				ProcessExecutionId:      latestPrcExe.ProcessExecutionId.String(),
+				ProcessStartedTimestamp: latestPrcExe.StartTimestamp,
+			},
+			ProcessType:            latestPrcExe.ProcessType,
+			RpcName:                request.GetRpcName(),
+			Input:                  request.Input,
+			LoadedGlobalAttributes: &loadGlobalAttrResp.Response,
+		},
+	).Execute()
+	if httpResp != nil {
+		defer httpResp.Body.Close()
+	}
+
+	if utils.CheckHttpResponseAndError(err, httpResp, s.logger) {
+		return nil, NewErrorWithStatus(
+			http.StatusFailedDependency,
+			"Failed to call worker RPC method. Error: "+err.Error()+" Http response: "+httpResp.Status)
+	}
+
+	err = utils.CheckDecision(resp.StateDecision)
+	if err != nil {
+		return nil, NewErrorWithStatus(
+			http.StatusBadRequest, err.Error())
+	}
+
+	updateResp, err := s.store.UpdateProcessExecutionFromRpc(ctx, data_models.UpdateProcessExecutionFromRpcRequest{
+		Namespace:          request.Namespace,
+		ProcessId:          request.ProcessId,
+		ProcessExecutionId: latestPrcExe.ProcessExecutionId,
+
+		StateDecision:       resp.GetStateDecision(),
+		PublishToLocalQueue: resp.GetPublishToLocalQueue(),
+
+		GlobalAttributeTableConfig: latestPrcExe.GlobalAttributeConfig,
+		UpdateGlobalAttributes:     resp.WriteToGlobalAttributes,
+
+		WorkerUrl:   latestPrcExe.WorkerUrl,
+		TaskShardId: persistence.DefaultShardId,
+	})
+	if err != nil {
+		return nil, s.handleUnknownError(err)
+	}
+	if updateResp.FailAtUpdatingGlobalAttributes {
+		s.logger.Warn("failed to update global attributes")
+		return nil, NewErrorWithStatus(
+			http.StatusFailedDependency,
+			"Failed to write global attributes, please check the error message for details: "+updateResp.UpdatingGlobalAttributesError.Error())
```

Comment:
Error handling is critical for building reliable applications. I think we need to return the error code to applications as mentioned in https://github.com/xdblab/xdb-apis/pull/24#discussion_r1393486766

Reply:
http.StatusFailedDependency means 424, this is the code.

Reply:
Why we need this error code -- 
the user code will make different decisions on this. If column value is conflict, or database schema is invalid, they should return different error code and different UI message in the application.

Reply:
You wrote it:
https://github.com/xdblab/xdb/blob/689585f839c05e29ffc0b3864e95fd002bd34b06/persistence/sql/complete_execute.go#L49

Reply:
This is in async state, we don’t have a good way to return error. We can discuss this in zoom

Reply:
And I also need to improve the error code in start process too to return error code

Reply:
> I meant the error code from the database driver.

The error code from the database driver should be returned by the `updateGlobalAttributesIfNeeded` method first. And here in the implementation for RPC, the `UpdatingGlobalAttributesError` is assigned to whatever returned from `updateGlobalAttributesIfNeeded`, and it's returned to SDK with `NewErrorWithStatus`.

So if you wanna imporve on the error message. you can just update `updateGlobalAttributesIfNeeded` itself in another PR. Currently, I will just return whatever the err is from the `updateGlobalAttributesIfNeeded` to SDK.